### ISSUE TITLE
Error when database or user name contain `-`

### ISSUE THREAD

**Issue by yujunz** [alexandrevilain/temporal-operator#402]
```
2023-06-27T11:46:08.436Z	[31mERROR[0m	Unable to create SQL database.	{"error": "pq: syntax error at or near \"-\"", "logging-call-at": "handler.go:94"}
```
Require quoting and escape in scripts.
```
    temporal-sql-tool
    --endpoint=postgres.aws-dev
    --port=5432 --user=temporal-aws-dev
    --password=$TEMPORAL_DEFAULT_DATASTORE_PASSWORD --database=temporal-aws-dev
    --plugin=postgres create
```

**Comment by alexandrevilain** [comment]
Hi! 
Thanks for the issue! I'll try to fix it as soon as possible. 
Maybe you're interested in fixing it ?

**Comment by yujunz** [comment]
New to the project. Any quick hint where the related code is? Particularly where is `temporal-sql-tool` from and how the job is created from CRD.

**Comment by yujunz** [comment]
And `temporal-sql-tool` seems from https://github.com/temporalio/temporal/blob/master/tools/sql/main.go IIUC.

**Comment by alexandrevilain** [comment]
Yep, you're right :)

**Comment by mobinanoorii** [comment]
can you assign this issue me?

**Comment by alexandrevilain** [comment]
Hi @mobinanoorii thanks but I think that @yujunz is trying to fix this :)

**Comment by yujunz** [comment]
I'm currently renaming it to `temporal_aws_dev` to work around. I doubt it requires some fix in `temporal-sql-tool` to support dash in the names.

You can claim it if you have time to fix it. @mobinanoorii

**Comment by alexandrevilain** [comment]
It think that the fix is to escape inputs in the bash script :)

**Comment by alexandrevilain** [comment]
Hi @mobinanoorii !

Do you need some help implementing this fix ? I'd like to create a release with this fix as soon as possible :)

**Comment by yujunz** [comment]
> And `temporal-sql-tool` seems from https://github.com/temporalio/temporal/blob/master/tools/sql/main.go IIUC.

Digging the code a bit. It seems formatting the connections string in format 

```
	dsnFmt     = "postgres://%v:%v@%v/%v?%v"
```

https://github.com/temporalio/temporal/blob/7efba9fb00d069491901b9d016ffb4dea02b3c16/common/persistence/sql/sqlplugin/postgresql/session/session.go#L40

https://github.com/temporalio/temporal/blob/7efba9fb00d069491901b9d016ffb4dea02b3c16/common/persistence/sql/sqlplugin/postgresql/session/session.go#L101

**Comment by alexandrevilain** [comment]
I think that it could be fixed by updating the following line: https://github.com/alexandrevilain/temporal-operator/blob/main/internal/resource/persistence/schema_scripts_configmap_builder.go#L131 

The fix could be something like: 

```
cmd = append(cmd, fmt.Sprintf(`--%s="%s"`, el.Key, el.Value))
```

But not 100% confident, it requires extended tests.

**Comment by alexandrevilain** [comment]
Released in 0.14.0

**Comment by yujunz** [comment]
Still the same error in `image: 'ghcr.io/alexandrevilain/temporal-operator:v0.14.0'`

```
2023-08-03T07:25:52.090Z	[31mERROR[0m	Unable to create SQL database.	{"error": "pq: syntax error at or near \"-\"", "logging-call-at": "handler.go:94"}
```

Shall we reopen the issue?

**Comment by alexandrevilain** [comment]
Yes, what's your user name ? I tried with "test-user" to fix this issue.

**Comment by yujunz** [comment]
```
    temporal-sql-tool
    --endpoint=postgres.aws-dev
    --port=5432 --user=temporal-aws-dev
    --password=$TEMPORAL_DEFAULT_DATASTORE_PASSWORD --database=temporal-aws-dev
    --plugin=postgres create
```

Both user and database contain `-`.

**Comment by alexandrevilain** [comment]
@yujunz I tried with your database name and I found the bug is in `temporal-sql-tool` itself. 
I created on PR on the temporal project, see: https://github.com/temporalio/temporal/pull/4735

**Comment by yujunz** [comment]
@alexandrevilain it seems the reviewers have requested some changes. Do you have time to address them?

**Comment by alexandrevilain** [comment]
Hi @yujunz ! 

Sorry for the delay! I forgot to answer, the requested change is not working. I answer now, waiting for their answer!

**Comment by alexandrevilain** [comment]
Hi @yujunz ! 

Seems this fix will be released in temporal v1.23.0! It's already available in v1.23.0-rc1 & v1.23.0-rc2.

Closing the issue :)
